### DISCUSSION TITLE
Controlling space between steps

### DISCUSSION THREAD

**Discussion by sylcha** [saadeghi/daisyui#1249] [Q&A] [answered]
Hi!
We are using delectably some DaisyUI components for our project. We have a question regarding the spacing between steps in the **steps** component. As you can see [here](https://coopmaths.fr/beta/?uuid=90c8c&id=can6a-2021&alea=cPh9&v=diaporama), it should display 30 steps but the last ones are hidden. We know that we have the possibility to use a scrollbar but for design purpose, this solution does not please us. I did not find a way in the documentation to control the spacing between steps : is there a way to achieve this anyway ?
Thanks in advance for your answer,
Sylvain

**Comment by saadeghi** [comment, marked as answer]
There are ways to customize the spacing but the result wouldn't look good. It would be 30 circles side by side. And on a mobile screen, it's even worse.  
I think for this design, if you don't want to use scrolls to show all the numbers, it's better to use vertical steps on a side bar. Or just show current step and next/prev buttons. What do you think?

**Reply by sylcha** [reply]
We opted for an automatic scroll when there is an overflow. It's handled with some javascript code. It does the job even if the last steps are not displayed: it's for projecting mental arithmetic slides so it could be important that students know about the numbers of questions though. May be we will work on some kind of dashed line as : ··· — 11 — 12 — 13 —    ···   — 29 — 30 but we aren't sure it's better...
```javascript
let steps: NodeListOf<HTMLLIElement>
if (stepsUl) steps = stepsUl.querySelectorAll("li")
if (steps && steps[currentQuestion + 5]) steps[currentQuestion + 5].scrollIntoView()
```
Thanks for your answer !
